### ISSUE TITLE
no-square image training? 

### ISSUE THREAD

**Issue by pure-rgb** [ultralytics/yolov5#5874]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Is the Yolo model resizing the input image to a square shape? 
Let's say if I have an input of shape (224, 480). Will it be able to train with this shape or the model will resize the image according to the target shape? I read the `train.py` but it's not clear to me.

**Comment by glenn-jocher** [comment]
@Suzan009 you can visualize batches directly by viewing your train batch jpgs. See **Train Custom Data** tutorial to get started:

![image](https://user-images.githubusercontent.com/26833433/144638845-37cc176d-cd10-4458-848a-df0d53983ad3.png)


## YOLOv5 Tutorials

* [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)&nbsp; ☘️ RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Supervisely Ecosystem](https://docs.ultralytics.com/yolov5/tutorials/model_export8)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training)
* [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading)&nbsp; ⭐ NEW
* [TorchScript, ONNX, CoreML Export](https://docs.ultralytics.com/yolov5/tutorials/model_export) 🚀
* [Test-Time Augmentation (TTA)](https://docs.ultralytics.com/yolov5/tutorials/test_time_augmentation)
* [Model Ensembling](https://docs.ultralytics.com/yolov5/tutorials/model_ensembling)
* [Model Pruning/Sparsity](https://docs.ultralytics.com/yolov5/tutorials/model_pruning_and_sparsity)
* [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution)
* [Transfer Learning with Frozen Layers](https://docs.ultralytics.com/yolov5/tutorials/transfer_learning_with_frozen_layers)&nbsp; ⭐ NEW
* [TensorRT Deployment](https://github.com/wang-xinyu/tensorrtx)

**Comment by pure-rgb** [comment]
@glenn-jocher
Thanks. It seems like yolo uses padding to make it square. Am I right? For example, in the below example, the up and down uses some padding to make it square, right? Like, if we do `python train.py --img 640 ... `, will it make our input to shape (640, 640)? 

![io](https://user-images.githubusercontent.com/45315076/144647474-0bc4c6d6-7e03-4d53-a6f8-d3ccda8fd630.png)

If yes. Then the above samples, even after padding doesn't look square to me. 


And another query regarding mosaic augmentation. Let's say we pick 4 samples. But after doing mosaic augmentation, one of the samples or all of them doesn't contain any target object (no bbox). See the below example, the lower part of the samples doesn't contain any bbox. Is it normal? 

![yui](https://user-images.githubusercontent.com/45315076/144648007-b9952f4f-754c-4947-a3ac-fc7730436628.png)

**Comment by glenn-jocher** [comment]
@Suzan009 yes everything is operating correctly.

**Comment by pure-rgb** [comment]
@glenn-jocher sorry, I think you don't get my concern properly. I want to know -  **will yolo-v5 training accept non-square input images? Or, it will resize the input image to a square shape??**

And for mosaic augmentation consists of 4 samples. Now, **after creating mosaic samples, how will it ensure that there would be enough bounding boxes of the final image??**

**Comment by glenn-jocher** [comment]
@Suzan009 YOLOv5 obviously accepts non-square images, otherwise the architecture would be nearly useless for real world applications. I don't know any standard camera format that outputs 1:1. You can see training images just by browsing COCO128 https://www.kaggle.com/datasets/ultralytics/coco128

EDIT: even your own screenshot shows non-square images.

**Comment by pure-rgb** [comment]
@glenn-jocher why is there's the gray background?

**Comment by glenn-jocher** [comment]
Imagenet mean value

**Comment by pure-rgb** [comment]
@glenn-jocher Thanks for being patient with me and answering my query. -)
